### PR TITLE
[Merged by Bors] - Avoid creating `prototype` property on methods

### PR DESCRIPTION
**Pull request by raskad**
This Pull Request changes the following:

- Stop creating a `prototype` property on class methods by passing a flag to the relevant opcodes.

### CODE REVIEW

File: boa_engine/src/bytecompiler/class.rs [hunk 1]
```diff
@@ -138,6 +138,7 @@ impl ByteCompiler<'_, '_> {
         let index = self.code_block.functions.len() as u32;
         self.code_block.functions.push(code);
         self.emit(Opcode::GetFunction, &[index]);
+        self.emit_u8(0);
```
Comment (nekevss) [inline boa_engine/src/bytecompiler/class.rs:141]: Thoughts on creating an `emit` method like `emit_with_opcode_and_u8` (maybe not the best name) to bind the pushing of the two operands together?

Comment (raskad) [reply boa_engine/src/bytecompiler/class.rs:141]: I think we could maybe generate `emit` functions for every opcode via a macro e.g. `Opcode::GetFunction::emit`. These functions could then automatically take the correct arguments.
Do we want to create a dedicated issue for making bytecode generation more typesafe?

**Review by nekevss — APPROVED** [review]
The rest looks good to me :)

File: boa_engine/src/bytecompiler/class.rs (hunk 1 above)
Comment (nekevss) [reply boa_engine/src/bytecompiler/class.rs:141]: Yeah, that would make sense to me.

**Review by jedel1043 — APPROVED** [review]
Looks good!

**Pull request closed without merging**